homeassistant/components/siren/__init__.py

```diff
@@ -0,0 +1,134 @@
+"""Component to interface with various sirens/chimes."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    SERVICE_TOGGLE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_ON,
+)
+from homeassistant.core import ServiceCall
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import (  # noqa: F401
+    PLATFORM_SCHEMA,
+    PLATFORM_SCHEMA_BASE,
+)
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.loader import bind_hass
+
+from .const import (
+    ATTR_AVAILABLE_TONES,
+    ATTR_DURATION,
+    ATTR_TONE,
+    ATTR_VOLUME_LEVEL,
+    DOMAIN,
+    SUPPORT_DURATION,
+    SUPPORT_TONES,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_SET,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+TURN_ON_SCHEMA = {
+    vol.Optional(ATTR_TONE): cv.string,
+    vol.Optional(ATTR_DURATION): cv.positive_float,
+    vol.Optional(ATTR_VOLUME_LEVEL): cv.small_float,
+}
+
+
+@bind_hass
+def is_on(hass: HomeAssistantType, entity_id: str) -> bool:
```

Comment:
Is this used?

Reply:
I added this because I saw it added to the humidity platform. If that is not required as a standard thing, we can remove it.

Reply:
I honestly think it is not used (or should not be used/get rid of in general).

But, this might be too much back in the history of Home Assistant to judge for me.

Reply:
I think this is mostly legacy.

We use it for some existing integrations, like device tracker and light in the device sun light trigger integration:
https://github.com/home-assistant/core/blob/d6ed5dac8b257f325c5793cf7d56c9eef2e2781b/homeassistant/components/device_sun_light_trigger/__init__.py#L113-L121


We also use it here, but that function doesn't seem used:
https://github.com/home-assistant/core/blob/d6ed5dac8b257f325c5793cf7d56c9eef2e2781b/homeassistant/components/__init__.py#L43

Reply:
then lets remove it 👍🏾